text/0000-floating-point-additional-precision.md

```diff
@@ -0,0 +1,192 @@
+- Feature Name: `allow_extra_fp_precision`
+- Start Date: 2019-04-08
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Update the Rust specification to allow floating-point operations to provide
+*more* precision than specified, but not less precision; this allows many safe
+optimizations. Specify robust mechanisms to disable this behavior.
+
+# Motivation
+[motivation]: #motivation
+
+Some platforms provide instructions to run a series of floating-point
+operations quickly, such as fused multiply-add instructions; using these
+instructions can provide performance wins up to 2x or more. These instructions
+may provide *more* precision than required by IEEE floating-point operations,
+such as by doing multiple operations before rounding or losing precision.
+Similarly, high-performance floating-point code could perform multiple
+operations with higher-precision floating-point registers before converting
+back to a lower-precision format.
+
+In general, providing more precision than required will only bring a
+calculation closer to the mathematically precise answer, never further away.
```

Comment:
That sounds like an extremely strong statement to me that needs further justification. I see no reason to assume such monotonicity here. Different rounding errors happening during a computation might as well just happen to cancel each other such that removing some errors actually *increases* the error of the final result.

Extreme example: `1 / 10` has a rounding error, but `1/10 - 1/10` actually gives the right result. Providing more precision only on one side of the subtraction *increases* the error of the entire calculation.

Reply:
See https://github.com/rust-lang/rfcs/pull/2686#discussion_r276418997